pkg/receive/handler.go

```diff
@@ -702,6 +667,58 @@ type retryState struct {
 	nextAllowed time.Time
 }
 
+type expectedErrors []*struct {
+	err   error
+	cause func(error) bool
+	count int
+}
+
+func (a expectedErrors) Len() int           { return len(a) }
+func (a expectedErrors) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a expectedErrors) Less(i, j int) bool { return a[i].count < a[j].count }
+
+// findCause returns a sentinel error that has occurred more than the given threshold.
+// It will inspect the error's cause if the error is a MultiError,
+// It will return cause of each contained error but will not traverse any deeper.
+func findCause(err error, threshold int) error {
```

Comment:
If I'm not mistaken there are exactly two types of calls to this function, with threshold being 1 or quorum. I think it would be useful to have distinct function names for the two cases, so as a caller I don't have to understand the parameter meanings.